src/app/rundown-view/components/minishelf/mini-shelf.component.ts

```diff
@@ -56,7 +56,7 @@ export class MiniShelfComponent implements OnInit, OnDestroy, OnChanges {
     if (!this.studioConfiguration) return
     if (this.media.duration < this.studioConfiguration.blueprintConfiguration.ServerPostrollDuration) return
 
-    this.mediaCalculatedDuration = this.media.duration - this.studioConfiguration.blueprintConfiguration.ServerPostrollDuration
+    this.mediaCalculatedDuration = Math.max(this.media.duration - this.studioConfiguration.blueprintConfiguration.ServerPostrollDuration, 0)
```

Comment:
Since you are now using `Math.max` the conditional on line 57 is no longer needed and should ideally be removed